plugins/custom-class/prism-custom-class.js

```diff
@@ -72,31 +69,26 @@
 	}
 
 	Prism.hooks.add('wrap', function (env) {
-		if (classAdders.length) {
-			/** @type {ClassAdderEnvironment} */
-			var adderEnv = {
+		if (adder) {
+			var result = adder({
 				content: env.content,
 				type: env.type,
 				language: env.language
-			};
-			for (var i = 0, l = classAdders.length; i < l; i++) {
-				var result = classAdders[i](adderEnv);
-				if (result) {
-					if (Array.isArray(result)) {
-						env.classes.push.apply(env.classes, result);
-					} else {
-						env.classes.push(result);
-					}
-				}
+			});
+
+			if (Array.isArray(result)) {
+				env.classes.push.apply(env.classes, result);
+			} else if (result) {
+				env.classes.push(result);
 			}
 		}
 
-		if (classMapper === defaultClassMap && !prefixString) {
+		if (!mapper && !prefixString) {
 			return;
 		}
 
 		env.classes = env.classes.map(function (c) {
-			return prefixString + classMapper(c, env.language);
+			return prefixString + mapper(c, env.language);
```

Comment:
I think this still needs a check for `mapper`'s existence. The check above only short-circuits if _both_ are undefined. But we could have no mapper but a prefix and we'll need to check this here.

Also, in reverse, if `prefixString` is undefined when we do this, `undefined` will get added as a prefix, so it also needs to be checked here.

Reply:
Ahh, right. My bad.